base/math.jl

```diff
@@ -413,19 +413,17 @@ and `val` is equal to ``x \\times 2^{exp}``.
 function frexp{T<:AbstractFloat}(x::T)
     xu = reinterpret(Unsigned,x)
     xe = xu & exponent_mask(T)
+    (xe == exponent_mask(T) || x == 0) && return x, 0 # +-0, NaN, or Inf
     k = Int(xe >> significand_bits(T))
     if xe == 0 # x is subnormal
-        x == 0 && return x, 0
```

Comment:
no reason to check x == 0 here within the xe==0. Moved these up top to reduce redundancy with the xe == 0 check.